lua/nvim-tree/lib.lua

```diff
@@ -4,7 +4,9 @@ local luv = vim.loop
 local renderer = require'nvim-tree.renderer'
 local config = require'nvim-tree.config'
 local git = require'nvim-tree.git'
-local diagnostics = require'nvim-tree.diagnostics'
+if vim.g.nvim_tree_lsp_diagnostics == 1 then
+  local diagnostics = require'nvim-tree.diagnostics'
+end
```

Comment:
sorry but this will error out when using `diagnostics.somefunc` because diagnostics will be nil outside of the `if` statement.

Reply:
I'm guarding the import here with the same guard you use throughout the rest of the file. All uses of and defs of `local diagnostics` are guarded in this file by `vim.g.nvim_tree_lsp_diagnostics`.